lib/reverse_words.rb

```diff
@@ -1,6 +1,34 @@
 # A method to reverse each word in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: linear, O(n) - where n is the length of the input string
+# Space complexity: constant, O(1) - the space used remains the same regardless
+#                   of input size.
 def reverse_words(my_words)
-  raise NotImplementedError
+  return if !my_words
+
+  start_index = 0
+  end_index = 0
+
+  while end_index < my_words.length
+    if my_words[end_index] == " " || end_index == my_words.length - 1
+      k = end_index
```

Comment:
A few nits:
1) Use a [ternary operator](https://www.w3resource.com/ruby/ruby-ternary-operator.php)? e.g. `k = my_words[end_index] == " " ? end_index - 1 : end_index`
2) Use a more self-explanatory variable name than `k`? (The other variable names are really good though! :+1: )